### PR TITLE
Document `asyncio` performance improvement in What's New 

### PR DESCRIPTION
**Pull request by kumaraditya303**

### CODE REVIEW

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -210,6 +210,11 @@ array
 asyncio
 -------
 
+* :mod:`asyncio` is optimized to avoid excessive copying when writing to socket
+  and uses :meth:`~socket.socket.sendmsg` if the platform supports it.
+  A speedup of 80% was observed on Linux on simple tcp echo server benchmark.
```
Comment (kumaraditya303) [inline Doc/whatsnew/3.12.rst:215]: Number is taken from `asyncio_tcp` benchmark from https://github.com/faster-cpython/ideas/blob/main/results/bm-20230402-3.12.0a6%2B-385b5d6/bm-20230402-linux-x86_64-python-385b5d6e091da454c3e0-3.12.0a6%2B-385b5d6-vs-3.10.4.md

Comment (kumaraditya303) [inline Doc/whatsnew/3.12.rst]: @AlexWaygood I welcome your suggestions to better document it.

File: Doc/whatsnew/3.12.rst [hunk 2]
```diff
@@ -210,6 +210,10 @@ array
 asyncio
 -------
 
+* :mod:`asyncio` is optimized to avoid excessive copying when writing to socket
+  and uses :meth:`~socket.socket.sendmsg` if the platform supports it.
+  (Contributed by Kumar Aditya in :gh:`91166`.)
```
Comment (AlexWaygood) [inline Doc/whatsnew/3.12.rst]: How about:

```suggestion
* The performance of writing to sockets in :mod:`asyncio` has been
  significantly improved. ``asyncio`` now avoids unnecessary copying when
  writing to sockets and uses :meth:`~socket.socket.sendmsg` if the platform
  supports it. (Contributed by Kumar Aditya in :gh:`91166`.)
```

Comment (kumaraditya303) [reply Doc/whatsnew/3.12.rst]: Thanks

**Pull request merged**